mne/preprocessing/ica.py

```diff
@@ -138,51 +141,58 @@ class ICA(ContainsMixin):
     ----------
     n_components : int | float | None
         Number of principal components (from the pre-whitening PCA step) that
-        are passed to the ICA algorithm during fitting. If :class:`int`, must
-        not be larger than ``max_pca_components``. If :class:`float` between 0
-        and 1, the number of components with cumulative explained variance
-        greater than ``n_components`` will be used. If ``None``,
-        ``max_pca_components`` will be used. Defaults to ``None``; the actual
-        number used when executing the :meth:`ICA.fit` method will be stored in
-        the attribute ``n_components_`` (note the trailing underscore).
-    max_pca_components : int | float | None
-        The number of principal components (from the pre-whitening PCA step)
-        that are retained for later use (i.e., for signal reconstruction in
-        `~mne.preprocessing.ICA.apply`; see the ``n_pca_components``
-        parameter). Use this parameter to reduce the dimensionality of the
-        input data via PCA before any further processing is performed.
-        If :class:`float` between 0 and 1, the number of components with
-        cumulative explained variance greater than ``max_pca_components`` will
-        be used. If ``1.0`` or ``None``, no dimensionality reduction occurs and
-        ``max_pca_components`` will equal the number of channels in the
-        `~mne.io.Raw` or `~mne.Epochs` object passed to
-        `~mne.preprocessing.ICA.fit`. Defaults to ``None``. When calling
-        `~mne.preprocessing.ICA.fit`, the attribute ``max_pca_components_``
-        will be set to the absolute number of retained PCA components.
+        are passed to the ICA algorithm during fitting:
+
+        - :class:`int`
+            Must be greater than 1 and less than or equal to
+            ``n_pca_components``.
+        - :class:`float` between 0 and 1 (exclusive)
+            The number of components with cumulative explained variance
+            of the data greater than ``n_components`` will be used. For
+            example, if 0.8 is passed then the number of components will
+            explain greater than 80%% of the variance.
+        - ``None``
+            ``n_pca_components`` or ``0.999999`` will be used, whichever
+            results in fewer components. This is done to avoid numerical
+            stability problems when whitening.
```

Comment:
In theory it can also happen when working with data with one huge first component, but I'll reword it to make it clear it's usually due to rank deficiency.